h/services/search_index.py

```diff
@@ -232,6 +235,28 @@ def sync(self, limit):
 
         return {key: len(value) for key, value in counts.items()}
 
+    def delete_annotations(self, limit):
+        """Delete annotations that've been deleted from the DB from Elasticsearch."""
+        jobs = self._queue_service.get(name="delete_annotation", limit=limit)
+
+        if not jobs:
+            log.info("There are not 'delete_annotation' jobs")
+            return
+
+        #: Completed jobs that can be removed from the queue.
+        completed_jobs = []
+
+        for job in jobs:
+            try:
+                annotation_id = job.kwargs["annotation_id"]
+                self.delete_annotation_by_id(annotation_id)
+                completed_jobs.append(job)
+                log.info("Deleted annotation %s from Elasticsearch", annotation_id)
+            except Exception:  # pylint:disable=broad-exception-caught
+                pass
+
+        self._queue_service.delete(completed_jobs)
```

Comment:
Rather than a separate `delete_annotations()` method, I'll probably add delete support into the [`SearchIndexService.sync()`](https://github.com/hypothesis/h/blob/4b90908e2fb7eb152fba4e05b5dc599bbb668519/h/services/search_index.py#L156-L240) method above: if the annotation to be synced either isn't in Postgres at all or has `deleted = true` in Postgres then `sync()` should delete the annotation from Elasticsearch.